src/controls/PrimitiveArrayControl.tsx

```diff
@@ -51,10 +54,12 @@ export function PrimitiveArrayControl({
     [addItem, path, schema, rootSchema],
   )
 
-  const prevDataValue = usePreviousValue(data as unknown)
-  if (data === undefined && prevDataValue === null) {
-    addDefaultItemToList()
-  }
+  const prevDataValue = usePreviousValue(data)
+  useEffect(() => {
```

Comment:
useEffect here is necessary to prevent console noise (timing of this update matters and useEffect causes it to happen outside of render loop)

Reply:
We had this and then removed it in final review. What console noise are we seeing?

Reply:
If you run tests, it says something about 'can't update state while rendering blah blah'